set.mm

```diff
@@ -742983,22 +742983,202 @@ to Davis and Putnam (1960).  (Contributed by David A. Wheeler,
 =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
   Implicational Calculus
 =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+
+  Implicational calculus is the fragment of propositional logic that uses only
+  material implication, and not negation. It can be axiomatized by inference
+  rule modus ponens ~ ax-mp together with the axioms { ~ ax-1 , ~ ax-2 ,
+  ~ peirce } or the Tarski-Bernays axioms { ~ ax-1 , ~ imim1 , ~ peirce } or
+  with the single axiom { ~ impsingle }. From either one of these three axiom
+  sets, all tautlogies containing only material implication may be proved. In
+  contrast, minimal implicational calculus, which is presented just before this
+  section, cannot prove certain tautologies ( ~ peirce , for example ). The
+  class of theorems proved by minimal implicational calculus is thus a subset
+  of the class of theorems proved by implicational calculus.
+
+  The primary source for this section is the paper by Lukasiewicz, _The
+  Shortest Axiom of the Implicational Calculus of Propositions_, Proceedings
+  of the Royal Irish Academy, Section A, vol. 52 (1948-1950), 25--33. It will
+  be cited as simply "Lukasiewicz" in the remainder of this section.
+
+  This section proves that the above three distinct axiom sets for
+  implicational calculus all produce the same class of theorems.
+
+  (Contributed by Larry Lesyna and Jeffrey P. Machado, 1-Aug-2023.)
+
 $)
 
-  $( Implicational calculus is the fragment of propositional logic that uses
-     only material implication, and not negation.  The shortest single axiom is
-     due to Lukasiewicz, see Lukasiewicz, <I>The Shortest Axiom of the
-     Implicational Calculus of Propositions</I>, Proceedings of the Royal Irish
-     Academy, Section A, vol. 52 (1948-1950), 25--33.  From this axiom and
-     ~ ax-mp , { ~ peirce , ~ ax-1 , ~ ax-2 } can be proved, and hence all
-     tautologies containing only material implication.  The axiom is proved
-     here starting from { ~ peirce , ~ ax-1 , and ~ ax-2 }.  Therefore, the
-     class of theorems proved by this single axiom, or alternatively {
-     ~ peirce , ~ ax-1 and ~ ax-2 } must be identical.  (Contributed by Larry
-     Lesyna and Jeffrey P. Machado, 18-Jul-2023.) $)
+
+  $( The shortest single axiom for implicational calculus, due to Lukasiewicz.
+     It is now known to be the _unique_ shortest axiom.  The axiom is proved
+     here starting from { ~ ax-1 , ~ ax-2 and ~ peirce }.  (Contributed by
+     Larry Lesyna and Jeffrey P. Machado, 18-Jul-2023.) $)
   impsingle $p |- ( ( ( ph -> ps ) -> ch )
      -> ( ( ch -> ph ) -> ( th -> ph ) ) ) $=
-    ( wi imim1 peirce a1d syl6 ) ABEZCECAEJAEZDAEJCAFKADABGHI $.
+    ( wi peirce a1d imim2 ax-mp imim1 ) CAEZABEZAEZEZKDAEZEZEZLCEZPEZMOEQMADABF
+    GMOKHIRNEQSELCAJRNPJII $.
```

Comment:
The existing proof is shorter and is also based on {ax-1, ax-2, peirce}, so no need to modify it. Actually, you should put both discouragement tags.

Reply:
Please tell me how to properly acknowledge this contribution.

Reply:
This was an automatic minimization (done with the "minimize_with *" command of metamath.exe) so it does not need acknowledgment tags. (The new proof is actually pretty close to yours: it only uses the lemma ~syl6 to shorten it somewhat.)